app/models/shipit/stack.rb

```diff
@@ -144,6 +144,8 @@ def merge_status
         'locked'
       else
         significant_statuses = undeployed_commits.map(&:significant_status)
+        significant_statuses << last_deployed_commit.significant_status unless last_deployed_commit.blank?
```

Comment:
Hum, unless I'm mistaken `undeployed_commits` is in reverse order (most recent first). So you probably want to `unshift` here.